kclvm/tools/src/query/override.rs

```diff
@@ -1,50 +1,90 @@
+use std::collections::{HashMap, HashSet};
+
+use anyhow::{anyhow, Result};
+
 use kclvm_ast::walker::MutSelfMutWalker;
 use kclvm_ast::{ast, walk_if_mut};
 use kclvm_parser::parse_expr;
 
-pub struct OverrideInfo {
-    pub pkgpath: String,
-    pub filename: String,
-    pub module: ast::Module,
-}
+use crate::printer::print_ast_module;
 
+/// Apply overrides on the AST program with the override specifications.
 pub fn apply_overrides(
     prog: &mut ast::Program,
-    overrides: &[ast::CmdOverrideSpec],
-    _import_paths: &[String],
-) {
+    overrides: &[ast::OverrideSpec],
+    import_paths: &[String],
+) -> Result<()> {
     for o in overrides {
         let pkgpath = if o.pkgpath.is_empty() {
             &prog.main
         } else {
             &o.pkgpath
         };
-        match prog.pkgs.get_mut(pkgpath) {
-            Some(modules) => {
-                for m in modules.iter_mut() {
-                    if fix_module_override(m, o) {}
-                    // module_add_import_paths(m, import_paths)
+        if let Some(modules) = prog.pkgs.get_mut(pkgpath) {
+            for m in modules.iter_mut() {
+                if apply_override_on_module(m, o, import_paths)? {
+                    let code_str = print_ast_module(m);
+                    std::fs::write(&m.filename, &code_str)?
                 }
             }
-            None => {}
         }
     }
+    Ok(())
 }
 
-pub fn fix_module_override(m: &mut ast::Module, o: &ast::CmdOverrideSpec) -> bool {
-    let ss = o.field_path.split(".").collect::<Vec<&str>>();
+/// Apply overrides on the AST module with the override specifications.
+pub fn apply_override_on_module(
+    m: &mut ast::Module,
+    o: &ast::OverrideSpec,
+    import_paths: &[String],
+) -> Result<bool> {
+    if !import_paths.is_empty() {
+        let mut exist_import_set: HashSet<String> = HashSet::new();
+        for stmt in &m.body {
+            if let ast::Stmt::Import(import_stmt) = &stmt.node {
+                exist_import_set.insert(import_stmt.path.to_string());
+            }
+        }
+        for (i, path) in import_paths.iter().enumerate() {
+            let line: u64 = i as u64 + 1;
+            if exist_import_set.contains(path) {
+                continue;
+            }
+            let name = path
+                .split('.')
+                .last()
+                .ok_or_else(|| anyhow!("Invalid import path {}", path))?;
+            let import_node = ast::ImportStmt {
+                path: path.to_string(),
+                rawpath: "".to_string(),
+                name: name.to_string(),
+                asname: None,
+            };
+            let import_stmt = Box::new(ast::Node::new(
+                ast::Stmt::Import(import_node),
+                m.filename.clone(),
+                line,
+                1,
```

Comment:
DO NOT hardcode index number like this. Abstract and maintain your indexing rules, which helps code readability and maintainability.

Reply:
change it to a const and leave a todo: abstract and maintain your indexing rules and open a new PR to solve it